### DISCUSSION TITLE
请问数据表格template字段支持排序吗？

### DISCUSSION THREAD

**Discussion by SouSuoKe** [layui/layui#2086] [问答] [answered]
![image](https://github.com/layui/layui/assets/27963195/70417086-070d-4a70-85f9-bbdbe6f16480)

如上图，差值是后端直接输出，差值状态 和 状态 是通过template判断差值输出的对应状态，现在测试发现这两列无法正常按状态排序，只会按整体数据的顺序正倒序。
请问是只能对后端返回的字段排序吗？template翻译出来的字段支持排序吗？

**Comment by Sight-wcg** [comment]
不支持的

**Comment by LSL1618** [comment, marked as answer]
你可以在拿到后端数据且在表格渲染之前，二次加工一下数据，预先准备好`差值状态`和`状态`这2个字段的值，这样就可以支持排序了。

**Comment by sentsim** [comment]
若非要支持，可设置 `autoSort: false`，采用服务端排序。
https://layui.dev/docs/2/table/#on-sort

**Reply by SouSuoKe** [reply]
感谢大佬 我后端加字段了
